addons/website/models/website.py

```diff
@@ -1837,6 +1838,24 @@ def _trigram_enumerate_words(self, search_details, search, limit):
                         value = value.lower()
                         yield from re.findall(match_pattern, value)
 
+    # As _trigram_enumerate_words use a hardcoded sql query, it needs some filter regarding performance. (Especially for is_published products)
+    def _trigram_enumerate_words_where_clause(self, fields):
+        if 'is_published' in fields and (self.env.user.has_group('base.group_portal') or self.env.user.has_group('base.group_public')):
+            return "WHERE is_published "
+        else:
+            return "WHERE True"
```

Comment:
Nitpicking:
```suggestion
            return "WHERE is_published"
        else:
            return "WHERE True"
```
or
```suggestion
            return "WHERE is_published "
        else:
            return "WHERE True "
```
But keep things coherent